geopandas/tools/geocoding.py

```diff
@@ -109,28 +109,21 @@ def reverse_geocode(points, provider='googlev3', **kwargs):
 
 
 def _query(data, forward, provider, **kwargs):
+    # generic wrapper for calls over lists to geopy Geocoders
     import geopy
-    from geopy.geocoders.base import GeocoderQueryError
+    from geopy.geocoders.base import Geocoder, GeocoderQueryError
+    from geopy.geocoders import get_geocoder_for_service
 
     if not isinstance(data, pd.Series):
         data = pd.Series(data)
 
-    # workaround changed name in 0.96
-    try:
-        Yahoo = geopy.geocoders.YahooPlaceFinder
-    except AttributeError:
-        Yahoo = geopy.geocoders.Yahoo
-
-    coders = {'googlev3': geopy.geocoders.GoogleV3,
-              'bing': geopy.geocoders.Bing,
-              'yahoo': Yahoo,
-              'openmapquest': geopy.geocoders.OpenMapQuest,
-              'nominatim': geopy.geocoders.Nominatim}
-
-    if provider not in coders:
-        raise ValueError('Unknown geocoding provider: {0}'.format(provider))
+    if isinstance(provider, str):
+        provider = get_geocoder_for_service(provider)
+    elif isinstance(provider, type):
+        if not issubclass(provider, Geocoder):
+            raise ValueError("provider must be a str or geopy Geocoder")
```

Comment:
`get_geocoder_for_service` raises a similar error to before for a string that is not supported. Since this change includes supporting a new argument type, I initially thought it might be worth checking for reasonable inputs. However, this check for a str or Geocoder instance is probably of marginal value, so not sure it is worth actually having.

Reply:
I think it is fine to raise here. 
But you can do with one level less I think (you don't need the `isinstance(provider, type`, as chekcing for subclass of Geocoder is enough?)

Reply:
would just add a test case for this error then

Reply:
`issubclass` requires a class as its first arguments, so the two level test is necessary. Will add a test though.

Reply:
Actually, in starting to write a test, and looking more into `issubclass`, it was hard to imagine the circumstance when that type checking would actually accomplish anything useful. If `issubclass` did support checking whether, say, a list was a subclass, it would have maybe been more valuable. So I removed this logic altogether.